robotbase/src/main/java/com/team2357/frc2022/arduino/ArduinoSensorState.java

```diff
@@ -0,0 +1,6 @@
+package com.team2357.frc2022.arduino;
+
+@FunctionalInterface
+public interface ArduinoSensorState {
```

Comment:
This should be more specific. Not all "SensorStates" will be true/false. Additionally, not all sensor states will be coming from an Arduino. I'd rename this to `SensorBooleanState` to be more descriptive.

Reply:
I'd also move this from the `arduino` package to somewhere more general. Keep in mind, we could possibly move one of our sensors over to the RoboRIO DIO itself, so whatever we do here needs to make sense in that context as well.